src/main/java/ar/com/fdvs/dj/core/DynamicJasperHelper.java

```diff
@@ -184,16 +189,52 @@ protected static DynamicJasperDesign generateJasperDesign(DynamicReport dr) thro
 		try {
 			if (dr.getTemplateFileName() != null) {
 				log.info("about to load template file: "+dr.getTemplateFileName() + ", Attemping to find the file directly in the file system.");
-				File file = new File(dr.getTemplateFileName());
-				if (file.exists()){
-					JasperDesign jdesign = JRXmlLoader.load(file);
-					jd = DJJRDesignHelper.downCast(jdesign,dr);
-				} else {
-					log.info("Not found: Attemping to find the file in the classpath...");
-					URL url = DynamicJasperHelper.class.getClassLoader().getResource(dr.getTemplateFileName());
-					JasperDesign jdesign = JRXmlLoader.load(url.openStream());
-					jd = DJJRDesignHelper.downCast(jdesign,dr);
-				}
+				boolean useJbossVirtuaFileSystem = false;
```

Comment:
Hi, in general it is not a good idea to couple a library to a specific technology unless totally necessary. 
What I can offer you is an implementation of `generateJasperDesign(DynamicReport dr)` with an extra parameter where you can pass an object that implements an interfase (lets name it DJTemplateLoader which has a single method:  `getTemplateFor(url) `and returns in `ImputSteam`

What do you think?